astropy/io/ascii/cds.py

```diff
@@ -143,7 +143,10 @@ def get_cols(self, lines):
                         for i in [1, 2, 3, 4]:
                             self.data.fill_values.append(('-' * i, fillval, col.name))
                     else:
-                        col.null = match.group('nullval')
+                        if (match.group('nullval') is None):
```

Comment:
A possible simplification that will avoid calling ``match.group('nullval')`` twice and minimize code change (isn't an expensive operation so not a problem but might help):

```suggestion
                        col.null = match.group('nullval')
                        if (col.null is None):
                            col.null = ''
```